rs/p2p/artifact_downloader/src/fetch_stripped_artifact/stripper.rs

```diff
@@ -16,24 +28,99 @@ impl Strippable for ConsensusMessage {
     type Output = MaybeStrippedConsensusMessage;
 
     fn strip(self) -> Self::Output {
+        let unstripped_consensus_message_id = self.id();
+
         match self {
             // We only strip data blocks.
             ConsensusMessage::BlockProposal(block_proposal)
                 if block_proposal.as_ref().payload.payload_type()
                     == ic_types::consensus::PayloadType::Data =>
             {
-                MaybeStrippedConsensusMessage::StrippedBlockProposal(block_proposal.strip())
+                let mut proto = pb::BlockProposal::from(&block_proposal);
+
+                // Remove the ingress payload from the proto.
+                if let Some(block) = proto.value.as_mut() {
+                    block.ingress_payload = None;
+                }
+
+                let data_payload = block_proposal.content.as_ref().payload.as_ref().as_data();
+                let stripped_ingress_payload = data_payload.batch.ingress.clone().strip();
```

Comment:
It seems a bit unfortunate that we have to clone all ingress messages here, when we actually want to get rid of them. I assume it's because only message above the threshold are stripped? 

I wonder if we could determine the size by comparing the `pos` of adjacent entries in `id_and_pos`, and then `get` the full message only if the size is below the threshold. But this is likely pre-mature optimization

Reply:
> It seems a bit unfortunate that we have to clone all ingress messages here, when we actually want to get rid of them. I assume it's because only message above the threshold are stripped?

It's actually because of how [Payload](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/types/types/src/consensus/payload.rs?L172) is defined. The inner payload is behind an Arc so technically there could be multiple owners of the data. I guess I could try to use [Arc::unwrap_or_clone](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.unwrap_or_clone) which will avoid cloning when possible but will clone more data than just the ingress payload (for example, the dkg payload), if we are not the only owners of the payload.

Reply:
I see, I guess technically you could imagine a function that creates a stripped ingress payload by only taking a reference to the original payload. But it might not be worth the effort